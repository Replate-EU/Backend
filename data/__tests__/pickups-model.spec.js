const db = require("../db-config");
const seed0 = require("../seeds/00-cleanup").seed;
const seed1 = require("../seeds/01-users").seed;
const seed2 = require("../seeds/02-volunteer_accounts").seed;
const seed3 = require("../seeds/03-business_accounts").seed;
const seed4 = require("../seeds/04-pickups").seed;
const pickupsDb = require("../pickups-model");

const newUser = {
  username: "darragh",
  password: "12345",
  user_type: "volunteer",
  contact_number: "1-234-567-78-90"
};

beforeAll(() => {
  const boop = Promise.all([
    seed0(db),
    seed1(db),
    seed2(db),
    seed3(db),
    seed4(db)
  ]);
  return boop;
});

describe("Updates pickups correctly", () => {
  it("updates the right pickup", async () => {
    const pickups = await db("pickups");
    const updated = await pickupsDb.update({ completed: true }, 1);
    const pickupsChanged = await db("pickups");
    expect(pickups).toMatchSnapshot();
    expect(pickupsChanged).toMatchSnapshot();
  });
});
