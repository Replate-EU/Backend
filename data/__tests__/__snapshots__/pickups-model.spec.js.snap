// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updates pickups correctly updates the right pickup 1`] = `
Array [
  Object {
    "business_id": 2,
    "claimed_by": 1,
    "completed": 0,
    "food_type": "doughnuts",
    "id": 1,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": 1,
    "completed": 1,
    "food_type": "fries",
    "id": 2,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "chicken",
    "id": 3,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "watermelon",
    "id": 4,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "cookies",
    "id": 5,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
]
`;

exports[`Updates pickups correctly updates the right pickup 2`] = `
Array [
  Object {
    "business_id": 2,
    "claimed_by": 1,
    "completed": 1,
    "food_type": "doughnuts",
    "id": 1,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": 1,
    "completed": 1,
    "food_type": "fries",
    "id": 2,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "chicken",
    "id": 3,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "watermelon",
    "id": 4,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
  Object {
    "business_id": 2,
    "claimed_by": null,
    "completed": 0,
    "food_type": "cookies",
    "id": 5,
    "pickup_time": 1580553507239,
    "quantity": 0.5,
  },
]
`;
